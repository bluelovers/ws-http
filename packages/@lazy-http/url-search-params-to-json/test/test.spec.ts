//@noUnusedParameters:false
/// <reference types="jest" />
/// <reference types="node" />
/// <reference types="expect" />

import { basename, extname } from 'path';
import {
	entriesToJSON,
	jsonToSearchParams,
	removeSquareBracketsFromJSON,
	searchParamsStringToJSON,
	searchParamsToJSON,
} from '../src/index';

beforeAll(async () =>
{

});

describe(basename(__filename, extname(__filename)), () =>
{

	test.skip(`dummy`, () => {});

	test.each([
		{ foo: ["first_value", "second_value"] },
		{
			foo: ['a', 'b', 'c'],
			bar: 'xyz'
		},
	] as const)('%j', (input) =>
	{
		//let input = { foo: ["first_value", "second_value"] } as const;

		let actual = jsonToSearchParams(input);

		let json = searchParamsToJSON(actual);

		expect(json).toStrictEqual(input)
		expect(entriesToJSON(actual.entries())).toStrictEqual(input)
		expect(entriesToJSON([...actual.entries()])).toStrictEqual(input)

		expect([...actual.entries()]).toMatchSnapshot();
		expect(actual.toString()).toMatchSnapshot();
		expect(json).toMatchSnapshot();

	});

	test(`squareBrackets`, () => {
		let input = { foo: ["first_value", "second_value"] } as const;

		let actual = searchParamsToJSON(jsonToSearchParams(input));
		let actual2 = searchParamsToJSON(jsonToSearchParams(input, {
			squareBracketsArrayKey: true,
		}));

		expect(actual).toStrictEqual(input);
		expect(actual).not.toStrictEqual(actual2);

		expect(actual2).toMatchSnapshot();

		let json2 = removeSquareBracketsFromJSON(actual2);

		expect(json2).toStrictEqual(actual);
		expect(jsonToSearchParams(actual2).toString()).toMatchSnapshot();

		expect(searchParamsStringToJSON(jsonToSearchParams(actual2).toString())).toStrictEqual(actual2);
		expect(searchParamsStringToJSON(jsonToSearchParams(actual2).toString().replace(/%5B%5D/g, '[]'))).toStrictEqual(actual2);

	})

})
