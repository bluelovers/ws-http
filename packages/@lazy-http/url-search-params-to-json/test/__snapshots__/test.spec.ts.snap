// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test.spec {"foo":["a","b","c"],"bar":"xyz"} 1`] = `
[
  [
    "foo",
    "a",
  ],
  [
    "foo",
    "b",
  ],
  [
    "foo",
    "c",
  ],
  [
    "bar",
    "xyz",
  ],
]
`;

exports[`test.spec {"foo":["a","b","c"],"bar":"xyz"} 2`] = `"foo=a&foo=b&foo=c&bar=xyz"`;

exports[`test.spec {"foo":["a","b","c"],"bar":"xyz"} 3`] = `
{
  "bar": "xyz",
  "foo": [
    "a",
    "b",
    "c",
  ],
}
`;

exports[`test.spec {"foo":["first_value","second_value"]} 1`] = `
[
  [
    "foo",
    "first_value",
  ],
  [
    "foo",
    "second_value",
  ],
]
`;

exports[`test.spec {"foo":["first_value","second_value"]} 2`] = `"foo=first_value&foo=second_value"`;

exports[`test.spec {"foo":["first_value","second_value"]} 3`] = `
{
  "foo": [
    "first_value",
    "second_value",
  ],
}
`;

exports[`test.spec squareBrackets 1`] = `
{
  "foo[]": [
    "first_value",
    "second_value",
  ],
}
`;

exports[`test.spec squareBrackets 2`] = `"foo%5B%5D=first_value&foo%5B%5D=second_value"`;
